openedx/core/djangoapps/timed_notification/tasks.py

```diff
@@ -0,0 +1,93 @@
+import logging
+
+from datetime import datetime, timedelta
+
+from core import send_course_notification_email, _get_course_link
+
+from celery.task import periodic_task
+from celery.schedules import crontab
+from lms.djangoapps.branding import get_visible_courses
+from pytz import utc
+from django.db import close_old_connections
+
+log = logging.getLogger('edx.celery.task')
+
+
+@periodic_task(run_every=crontab(minute=1, hour=0))
+def execute_dialy():
+    log.info("TIMED-NOTIFICATION-JOB execute_dialy")
+
+    close_old_connections()
+    courses = get_visible_courses()
+
+    while courses:
+        course = courses[-1]
+
+        # we're interested only in dates
+        course_start_date = course.start.date()
+        date_now = datetime.now(utc).date()
+
+        # check if course is already started
+        if date_now > course_start_date:
+            courses.pop()
+            continue
+
+        context = {
+            'course_name': course.display_name,
+            'course_link': _get_course_link(course_id=course.id),
+        }
+
+        # send email when 7 days left to course start
+        if course_start_date - timedelta(days=7) == date_now:
+
+            send_course_notification_email(course=course,
+                                           mako_template_path='timed_notification/course_early_welcome.txt',
+                                           context=context)
+        # send email when 2 days left to course start
+        elif course_start_date - timedelta(days=2) == date_now:
+            send_course_notification_email(course=course,
+                                           mako_template_path='timed_notification/course_start_reminder.txt',
+                                           context=context)
+        # send email the day the course starts
+        elif course_start_date == date_now:
+            send_course_notification_email(course=course,
+                                           mako_template_path='timed_notification/course_welcome.txt',
+                                           context=context)
+        courses.pop()
+
+
+@periodic_task(run_every=crontab(minute=1, hour=0, day_of_week=1))
```

Comment:
I think this job can be eliminated and we can use job above checking of today = Monday